pkg/logger/logger.go

```diff
@@ -0,0 +1,67 @@
+package logger
+
+import (
+	"os"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var logLevelMapping = map[string]int{
+	"devel":   0,
+	"default": 1, // default one when not set log-mode
+	"prod":    2,
+}
+
+// in each controller, to use different log level.
+func LogWithLevel(logger logr.Logger, level string) logr.Logger {
+	if len(level) == 0 {
```

Comment:
Probably, there is no sense for the check, the lookup check does the same.
Or may be make it non-mandatory argument (...string) and then check for the first member. But since it's an internal API, does not make a lot of sense.

Reply:
you are right.
the original idea is to ensure, any " " will be chopped but i made a wrong code.
i did a bit change now to cover the cases 
- `"   devel" `is the same as `"devel"`, 
- `"          " `is the same as `""` as `"default"`